src/main/java/com/amazonaws/kinesisvideo/demoapp/activity/WebRtcActivity.java

```diff
@@ -259,6 +265,39 @@ private boolean isValidClient() {
         return client != null && client.isOpen();
     }
 
+    private void handlePendingIceCandidates(String clientId) {
+        // Check if ICE candidates are available in the queue and add the candidate
+        while(!pendingIceCandidatesQueue.isEmpty()) {
+            final Event message = pendingIceCandidatesQueue.peek();
+            final IceCandidate iceCandidate = Event.parseIceCandidate(message);
+            if(clientId == message.getSenderClientId()) {
+                final boolean addIce = localPeer.addIceCandidate(iceCandidate);
+                Log.d(TAG, "Added ice candidate after sdp exchange " + iceCandidate + " " + (addIce ? "Successfully" : "Failed"));
+            }
+            pendingIceCandidatesQueue.remove();
+        }
+    }
+
+    private void checkAndAddIceCandidate(Event message, IceCandidate iceCandidate) {
```

Comment:
Not entirely following the logic here.

Reply:
I have added some comments to clarify. The logic is:

1. If SDP exchange is not completed yet, we update the peerConnection found map to reflect that peer connection is not found for the specified client ID and then update the hash map for pending ice candidates
2. If we received subsequent ICE candidates before receiving offer/answer, we just update the existing  pending queue entry in the map
3. If peer connection is established, we just add the received ice candidate